mlos_core/mlos_core/optimizers/bayesian_optimizers.py

```diff
@@ -276,6 +277,177 @@ def surrogate_predict(self, configurations: pd.DataFrame, context: Optional[pd.D
         return ret
 
     def acquisition_function(self, configurations: pd.DataFrame,
-                             context: Optional[pd.DataFrame] = None) -> Callable:
+                             context: Optional[pd.DataFrame] = None) -> npt.NDArray:
         # This seems actually non-trivial to get out of skopt, so maybe we actually shouldn't implement this.
         raise NotImplementedError()
+
+
+class SmacOptimizer(BaseBayesianOptimizer):
```

Comment:
Cosmetic, but for maintenance, I'm thinking we should split this file out to have each class in its own file first.  We should tackle that in a separate PR though.

Reply:
Maybe do that first ...

Reply:
See: #342

Reply:
#342 should go in shortly.  Please rebase off of that.  Thanks!